dts: bindings: gpio: use hyphens instead of underscore

jilaypandya opened the pull request:
replace underscore with hyphens to comply with device tree spec

Todos:
- [x] Add migration guide entry commit

jilaypandya committed:
dts: bindings: gpio: use hyphens instead of underscore

replace underscore with hyphens to comply with device tree spec

Signed-off-by: Jilay Pandya <jilay.pandya@outlook.com>

kartben reviewed (changes requested):
needs migration guide entry

jilaypandya committed:
docs: migration_guide_4.1: entry for renamed device tree properties

add migration guide after renaming device tree properties
- pin_mask -> pin-mask
- pinmux_mask -> pinmux-mask
- vbatts_pins -> vbatts-pins
- bit_per_gpio -> bit-per-gpio
- off_val -> off-val
- on_val -> on-val

Signed-off-by: Jilay Pandya <jilay.pandya@outlook.com>

kartben commented on doc/releases/migration-guide-4.1.rst:
```diff
@@ -170,6 +170,16 @@ Entropy
+GPIO
+====
+
+* Renamed the device tree property ``pin_mask`` to ``pin-mask``.
+* Renamed the device tree property ``pinmux_mask`` to ``pinmux-mask``.
+* Renamed the device tree property ``vbatts_pins`` to ``vbatts-pins``.
+* Renamed the device tree property ``bit_per_gpio`` to ``bit-per-gpio``.
+* Renamed the device tree property ``off_val`` to ``off-val``.
+* Renamed the device tree property ``on_val`` to ``on-val``.
```
`s/device tree/Devicetree/g` but this can be addressed at a later stage when cleaning up release documentation closer to release date.
